src/client/content_mapblock.cpp

```diff
@@ -713,8 +713,9 @@ void MapblockMeshGenerator::drawLiquidSides()
 			if (data->m_smooth_lighting)
 				cur_node.color = blendLightColor(pos);
 			pos += cur_node.origin;
-			vertices[j] = video::S3DVertex(pos.X, pos.Y, pos.Z, 0, 0, 0, cur_node.color, vertex.u, v);
+			vertices[j] = video::S3DVertex(pos.X, pos.Y, pos.Z, face.dir.X, face.dir.Y, face.dir.Z, cur_node.color, vertex.u, v);
```

Comment:
For note reflections?

Reply:
Indeed. It seems strange to me that this wasn’t done to begin with… I don’t think this change makes anything look particularly different save for the reflections, or maybe it also breaks something I’m not aware of… But if this is not present, then flowing water looks irritatingly bright, or you would have to remove reflections from it in which case it looks irritatingly dark.

Reply:
setting correct normals shouldn't break anything :)